lib/prelude.dx

```diff
@@ -2469,3 +2535,15 @@ instance {a b} [Ix a, Ix b] Ix (a=>b)
   size = intpow (size b) (size a)
   ordinal             = reversed_digits_to_int
   unsafe_from_ordinal = int_to_reversed_digits
+
+instance {a b} [NonEmpty a, NonEmpty b] NonEmpty (a=>b)
+  first_ix = unsafe_from_ordinal _ 0
+
+instance {a b} [NonEmpty a, Ix b, Subset a b] NonEmpty b
+  first_ix = unsafe_from_ordinal _ 0
+
+instance {a b} [NonEmpty b, Eq a, Eq b] Subset a (b=>a)
```

Comment:
This feels very arbitrary too. I think the `Subset` is mainly meant for sum-type-like things, not for product types.

Reply:
I agree that the injection strategy is a bit arbitrary, I'll remove.  I got carried away with all the wonderful things I can prove with `NonEmpty` instances!